File: home_assignments/csharp/TicketSeller/model/validators/implementations/TicketValidator.cs
using System;
using model.entities;
using model.validators.interfaces;

namespace model.validators.implementations;

public class TicketValidator : ITicketValidator
{
    public void Validate(Ticket ticket)
    {
        String buyerName = ticket.BuyerName;
        int spots = ticket.NumberOfSpots;
        Festival festival = ticket.Festival;
        String error = "";
        if (buyerName == null)
            error += "buyerName is null;\n";
        else
        {
            if (buyerName.Equals(""))
            {
                error += "Buyer Name should be non empty!;\n";
            }
        }

        if (spots == null)
            error += "spots is null;\n";
        else
        {
            if (spots <= 0)
            {
                error += "Spots should be a positive number!;\n";
            }

            if (spots > festival.RemainingSpots)
            {
                error += "Not enough available spots!;\n";
            }
        }

        if (!error.Equals(""))
        {
            throw new ValidationException(error);
        }
    }
}

File: home_assignments/java/TicketSeller/TicketSellerModel/src/main/java/ticketseller/model/validators/implementations/TicketValidator.java
package ticketseller.model.validators.implementations;


import ticketseller.model.entities.Festival;
import ticketseller.model.entities.Ticket;
import ticketseller.model.validators.ValidationException;
import ticketseller.model.validators.interfaces.ITicketValidator;

public class TicketValidator implements ITicketValidator {
    @Override
    public void validate(Ticket ticket) throws ValidationException {
        String buyerName = ticket.getBuyerName();
        Integer spots = ticket.getNumberOfSpots();
        Festival festival = ticket.getFestival();
        String error = "";
        if (buyerName == null)
            error += "buyerName is null;\n";
        else {
            if (buyerName.equals("")) {
                error += "Buyer Name should be non empty!;\n";
            }
        }
        if (spots == null)
            error += "spots is null;\n";
        else {
            if (spots <= 0) {
                error += "Spots should be a positive number!;\n";
            }
            if (spots > festival.getRemainingSpots()) {
                error += "Not enough available spots!;\n";
            }
        }
        if (!error.equals("")) {
            throw new ValidationException(error);
        }
    }
}
